Assets/Packages/com.studio23.ss2.splashscreensystem/Runtime/UI/SplashScreenUIManager.cs

```diff
@@ -0,0 +1,118 @@
+                imageObjects.Add(newImage);
+                rectParent.GetComponent<GridLayoutGroup>().cellSize = new Vector2(entry.image.width, entry.image.height);
+            }
+        }
+
+        public async void CrossFadeData(float duration)
+        {
+            float val = 0.0f;
+            ParentPanel.GetComponent<CanvasGroup>().alpha = val;
+            while (val < duration)
+            {
+                val += Time.deltaTime;
+                var ratio = val / duration;
+                ParentPanel.GetComponent<CanvasGroup>().alpha = ratio;
+                await UniTask.Yield();
+            }
+        }
+
+        public void DisplayData(ScriptableObject data)
+        {
+            _splashData = data;
+
+            if (data is ScreenTextData)
+            {
+                ScreenTextData textData = data as ScreenTextData;
+                if (textData != null)
+                {
+                    titleText.text = textData.Title;
+                    ScrollRect.GetComponentInChildren<TextMeshProUGUI>().text = textData.Description;
+                    titleText.gameObject.SetActive(true);
+                    ScrollRect.gameObject.SetActive(true);
+                    rectParent.gameObject.SetActive(false);
+                    ScrollRect.GetComponent<ScrollRect>().vertical = textData.ShowButton;
+                    ButtonPanel.SetActive(textData.ShowButton);
+                    AcceptBtn.onClick.RemoveAllListeners();
+                    DeclineBtn.onClick.RemoveAllListeners();
+                    AcceptBtn.onClick.AddListener(() =>
+                    {
+                        SplashScreenBehaviour.OnSubmit(true);
+                        SplashScreenBehaviour.HideSplashScreen();
+                    });
+                    DeclineBtn.onClick.AddListener(() =>
+                    {
+                        SplashScreenBehaviour.OnSubmit(false);
+                        SplashScreenBehaviour.HideSplashScreen();
+                    });
+                }
+            }
+            else if (data is ThirdPartyData)
+            {
+                ThirdPartyData thirdPartyData = data as ThirdPartyData;
+
+                if (thirdPartyData != null && thirdPartyData.ThirdPartyEntries.Count > 0)
+                {
+                    PopulateUI();
+
+                    Vector2 pivot = new Vector2(0.5f, 0.5f);
+
+                    ThirdPartyEntry entry = thirdPartyData.ThirdPartyEntries[0];
+                    titleText.text = entry.title;
+                    ScrollRect.GetComponentInChildren<TextMeshProUGUI>().text = "";
+                    ButtonPanel.SetActive(false);
+                    titleText.gameObject.SetActive(false);
+                    ScrollRect.gameObject.SetActive(false);
+                    rectParent.gameObject.SetActive(true);
+                }
```

Comment:
Would recommend virtual functions instead of casting scriptableobject to each type and checking it.